src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/adapter/WebClientUtils.kt

```diff
@@ -0,0 +1,100 @@
+package uk.gov.justice.digital.hmpps.manageusersapi.adapter
+
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.util.UriBuilder
+import java.net.URI
+
+class WebClientUtils(private val client: WebClient) {
+
+  fun <T : Any> get(uri: String, elementClass: Class<T>): T =
+    client.get()
+      .uri(uri)
+      .retrieve()
+      .bodyToMono(elementClass)
+      .block()!!
+
+  fun <T : Any> getIfPresent(uri: String, elementClass: Class<T>): T? =
+    client.get()
```

Comment:
Because it's T? then we can just declare as <T> or  <T: Any?>

Reply:
discussed & leaving as is